lib/pbio/drv/uart/uart_ev3_pru_lib/suart_api.c

```diff
@@ -2319,168 +1520,83 @@ int16_t arm_to_pru_intr_init(void) {
-            return -1;
-        }
+    /* Sets the channel for the system interrupt
+    * MAP channel 0 to SYS_EVT32
+    * MAP channel 1 to SYS_EVT33
+    * MAP channel 2 to SYS_EVT34  SUART0-Tx
+    * MAP channel 2 to SYS_EVT35  SUART0-Rx
+    */
+    u32offset =
+        (uint32_t)pru_arm_iomap.pru_io_addr | (PRU_INTC_CHANMAP8 &
+            0xFFFF);
+    u32value = 0x02020100;
+    s16retval =
+        pru_ram_write_data_4byte(u32offset, (uint32_t *)&u32value, 1);
+    if (-1 == s16retval) {
+        return -1;
     }
 
-    if ((PRU0_MODE == PRU_MODE_RX_ONLY) || (PRU1_MODE == PRU_MODE_RX_ONLY) ||
-        (PRU0_MODE == PRU_MODE_TX_ONLY) || (PRU1_MODE == PRU_MODE_TX_ONLY)) {
-        /* Sets the channel for the system interrupt
-        * MAP channel 0 to SYS_EVT32
-        * MAP channel 1 to SYS_EVT33
-        * MAP channel 2 to SYS_EVT34  SUART0
-        * MAP channel 3 to SYS_EVT35  SUART1
-        */
-        u32offset =
-            (uint32_t)pru_arm_iomap.pru_io_addr | (PRU_INTC_CHANMAP8 &
-                0xFFFF);
-        u32value = 0x03020100;
-        s16retval =
-            pru_ram_write_data_4byte(u32offset, (uint32_t *)&u32value, 1);
-        if (-1 == s16retval) {
-            return -1;
-        }
-
-        /* Sets the channel for the system interrupt
-        * MAP channel 4 to SYS_EVT36	SUART2
-        * MAP channel 5 to SYS_EVT37    SUART3
-        * MAP channel 6 to SYS_EVT38    SUART4
-        * MAP channel 7 to SYS_EVT39    SUART5
-        */
-        u32offset =
-            (uint32_t)pru_arm_iomap.pru_io_addr | (PRU_INTC_CHANMAP9 &
-                0xFFFF);
-        u32value = 0x07060504;
-        s16retval =
-            pru_ram_write_data_4byte(u32offset, (uint32_t *)&u32value, 1);
-        if (-1 == s16retval) {
-            return -1;
-        }
+    /* Sets the channel for the system interrupt
+    * MAP channel 3 to SYS_EVT36	SUART1-Tx
+    * MAP channel 3 to SYS_EVT37    SUART1-Rx
+    * MAP channel 4 to SYS_EVT38    SUART2-Tx
+    * MAP channel 4 to SYS_EVT39    SUART2-Rx
+    */
+    u32offset =
+        (uint32_t)pru_arm_iomap.pru_io_addr | (PRU_INTC_CHANMAP9 &
+            0xFFFF);
+    u32value = 0x04040303;
+    s16retval =
+        pru_ram_write_data_4byte(u32offset, (uint32_t *)&u32value, 1);
+    if (-1 == s16retval) {
+        return -1;
+    }
 
-        /* Sets the channel for the system interrupt
-        * MAP channel 8 to SYS_EVT40	SUART6
-        * MAP channel 9 to SYS_EVT41    SUART7
-        * MAP channel 2 to SYS_EVT42    SUART0
-        * MAP channel 3 to SYS_EVT43    SUART1
-        */
-        u32offset =
-            (uint32_t)pru_arm_iomap.pru_io_addr | (PRU_INTC_CHANMAP10 &
-                0xFFFF);
-        u32value = 0x03020908;
-        s16retval =
-            pru_ram_write_data_4byte(u32offset, (uint32_t *)&u32value, 1);
-        if (-1 == s16retval) {
-            return -1;
-        }
+    /* Sets the channel for the system interrupt
+    * MAP channel 5 to SYS_EVT40	SUART3-Tx
```

Comment:
```suggestion
    * MAP channel 5 to SYS_EVT40    SUART3-Tx
```

tabs vs. spaces

Reply:
There is also some line wrapping that could be improved on lines we change, but don't spend too much time on it.

Reply:
I made a reasonable effort to fix the most wonky of the line wrapping.